lays initial foundation for discord bot work

grishrl opened the pull request:
Copied over the code for loading in from files for all commands / events we want to use
refactored the web-api stuff I started with earlier today, 
we will have a folder for web-api, commands, and events
we can use this foundation to begin creating the discord bot

Commit:
install discord.js, express, some http parsers, provide an URI for creating a role and assigning user
create ground level sandbox for building up discord bots for NGS

Commit:
update readme since dependencies have been added

Commit:
adds config file to manage permissions and discord client config
update index.js to load commands and actions from file to lay foundation for continued work
refactor upsertURI code - found a bug preventing running
add several 'command' files that we can prune or edit later
add several 'event' files as a foundation to edit or prune later
add helper functions for this foundation
add Logger functions but ended up adding a 3rd party logger possibly remove later

Commit:
remove logger from package and use included

Commit:
add folder for web-api calls

Commit:
update readme with deps

carlosplusplus commented on README.md:
```diff
@@ -1,2 +1,24 @@
+## Getting Started
+
+### Prerequisites
+- Node.js and NPM installation, you may need some extra peices for a few dependencies namely enmap check here for more (https://enmap.evie.codes/install)
```
🎨- `pieces` is misspelled.

grishrl replied:
got it chief

carlosplusplus commented on README.md:
```diff
@@ -1,2 +1,24 @@
 Custom bot, built in discord.js, for use with NGS' Discord server.
+
+## Getting Started
+
```
I'd love to use `nvm` here and lock the project to a specific version of `npm` and `node`.

grishrl replied:
I have added node to the dependencies of the package - what will require having said node version and will cause npm to install that when running the install package; (however user will need node & npm (come together) in order to get a toe in the water)
From what I can tell nvm is just an environment manager to switch between node versions, am I missing something - I don't have nvm installed.

carlosplusplus replied:
Yup, that's exactly right - `nvm` ensures that every single person running on this project is running the same version of `node` and `npm`. Without control there, you're going to run into situations where `npm install` might generate noise / garbage in `package.json` that varies from person to person, and now PRs become polluted.

Super easy to install - I'd pick the latest and greatest.

We can also consider using `yarn` over `npm`.

grishrl replied:
Well I've been mulling it over today and I'm still a bit skeptical about the gains of doing this --
From what I can tell the nvm is a great tool especially for mac/linux users but from what i've gathered the nvmrc file doesn't work with the windows style nvm, it seems like it's a nice utility tool for folks who want to use nvm (sort of like a rb-env) but forcing it into the project doesn't seem like the right play to me.  
Further if we want to use yarn in leu or along side of npm then that should be fine I've seen lots of repos using both but I don't see any clear reason do one instead of the other
In short I don't fully see the need or gain from implementing another layer of complexity to the repo as lots of repos run with only node and npm, the package and package-lock handle the dependency tracking and it should only grow when we add new dependencies

carlosplusplus replied:
All fine - if we run into any issues with package installation or `npm` specific changes to `package.json` due to specific versions (which is what this tries to address), we can revisit this later.

Commit:
update readme and package files

Commit:
adds jimp to package for image processing
adds utils file for extra helper functions
add embed generator function
add authorize middleware to ensure api caller has permission
add api for receiving schedules to create embeds for

Commit:
shaking the dust off this bot; enhancing the send messages to channel feature; getting it up to snuff for prime time

Commit:
update embed generator to properly build a match link

Commit:
payload example
